cpp/src/Slice/Parser.cpp

```diff
@@ -2102,1144 +1933,1492 @@ Slice::Container::lookupException(const string& scoped, bool printError)
-        ClassDefPtr q = ClassDefPtr::dynamicCast(*p);
-        if(q)
+        // Find the referenced scope
+        ContainerPtr container = const_cast<Container*>(this);
+        string scope = scoped.substr(0, scoped.rfind("::"));
+        ContainedList cl = container->lookupContained(scope, false);
+        if(!cl.empty())
         {
-            result.push_back(q);
+            if(ModulePtr module = ModulePtr::dynamicCast(cl.front()))
+            {
+                EnumList enums = module->enums();
+                string name = scoped.substr(lastColon + 1);
+                for (EnumList::iterator p = enums.begin(); p != enums.end(); ++p)
+                {
+                    ContainedList cl2 = (*p)->lookupContained(name, false);
+                    if(!cl2.empty())
+                    {
+                        result.push_back(EnumeratorPtr::dynamicCast(cl2.front()));
+                    }
+                }
+            }
         }
     }
+
     return result;
 }
 
-ExceptionList
-Slice::Container::exceptions() const
+bool
+Slice::Container::hasContentsWithMetaData(const string& meta) const
 {
-    ExceptionList result;
-    for (ContainedList::const_iterator p = _contents.begin(); p != _contents.end(); ++p)
+    for (const auto& content : contents())
     {
-        ExceptionPtr q = ExceptionPtr::dynamicCast(*p);
-        if(q)
+        if(content->hasMetaData(meta))
         {
-            result.push_back(q);
+            return true;
         }
-    }
-    return result;
-}
 
-StructList
-Slice::Container::structs() const
-{
-    StructList result;
-    for (ContainedList::const_iterator p = _contents.begin(); p != _contents.end(); ++p)
-    {
-        StructPtr q = StructPtr::dynamicCast(*p);
-        if(q)
+        ContainerPtr container = ContainerPtr::dynamicCast(content);
+        if(container && container->hasContentsWithMetaData(meta))
         {
-            result.push_back(q);
+            return true;
         }
     }
-    return result;
+
+    return false;
 }
 
-SequenceList
-Slice::Container::sequences() const
-{
-    SequenceList result;
-    for (ContainedList::const_iterator p = _contents.begin(); p != _contents.end(); ++p)
+string
+Slice::Container::thisScope() const
+{
+    string s;
+    ContainedPtr contained = ContainedPtr::dynamicCast(const_cast<Container*>(this));
+    if(contained)
     {
-        SequencePtr q = SequencePtr::dynamicCast(*p);
-        if(q)
-        {
-            result.push_back(q);
-        }
+        s = contained->scoped();
     }
-    return result;
+    s += "::";
+    return s;
 }
 
-DictionaryList
-Slice::Container::dictionaries() const
+void
+Slice::Container::sort()
 {
-    DictionaryList result;
-    for (ContainedList::const_iterator p = _contents.begin(); p != _contents.end(); ++p)
-    {
-        DictionaryPtr q = DictionaryPtr::dynamicCast(*p);
-        if(q)
-        {
-            result.push_back(q);
-        }
-    }
-    return result;
+    contents().sort();
```

Comment:
What's the point of sorting a temporary variable? 
This suggests this `sort` function is never called.

Reply:
Removed.